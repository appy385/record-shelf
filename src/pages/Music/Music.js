import React, { useEffect, useState } from 'react';
import { getSongs } from '../../utils/api';
import Song from '../../components/Song/Song';
import iconGenre from '../../assets/icon-genre.svg';
import iconGrid from '../../assets/icon-grid.svg';
import './Music.scss';
import groupByGenre from '../../utils/helper';
import Genre from '../../components/Genre/Genre';

const Music = () => {
  const [songs, setSongs] = useState([]);
  const [genreSongs, setGenreSongs] = useState({});
  const [isGenre, setIsGenre] = useState(false);
  useEffect(async () => {
    const songList = await getSongs();
    const newGenreSongs = groupByGenre(songList.data);
    setGenreSongs(newGenreSongs);
    setSongs(songList.data);
  }, []);
  const toggle = () => {
    setIsGenre(!isGenre);
  };
  return (
    <div className="music">
      {(isGenre) ? (
        <>
          <div className="music-header-container">
            <span className="music-header-message">Genre</span>
            <input type="image" className="music-toggle-button" onClick={toggle} src={iconGrid} alt="text" />
          </div>
          <div className="music-container">
            <Genre genreSongs={genreSongs} />
          </div>
        </>
      )
        : (
          <>
            <div className="music-header-container">
              <span className="music-header-message">All Songs</span>
              <input type="image" className="music-toggle-button" onClick={toggle} src={iconGenre} alt="text" />
            </div>
            <div className="music-container">
              {songs.map((song) => (<Song key={song.id} song={song} />))}
            </div>
          </>
        )}

    </div>
  );
};

export default Music;
