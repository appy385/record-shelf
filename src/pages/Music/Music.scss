.music {
    margin: 6rem;
}
.music-allsongs {
    font-size: 3rem;
    font-weight: 700;
}
.music-container {
    display: flex;
    flex-wrap: wrap;
   

}

