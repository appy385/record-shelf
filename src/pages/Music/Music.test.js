import React from 'react';
import { render, screen } from '@testing-library/react';
import Music from './Music';

describe(Music.name, () => {
  test('should match snapshot', () => {
    const { container } = render(<Music />);
    expect(container).toMatchSnapshot();
  });
  test('should display text All songs', () => {
    render(<Music />);
    screen.getByText('All Songs');
  });
});
