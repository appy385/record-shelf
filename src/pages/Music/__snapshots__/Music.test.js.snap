// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music should match snapshot 1`] = `
<div>
  <div
    class="music"
  >
    <div
      class="music-header-container"
    >
      <span
        class="music-header-message"
      >
        All Songs
      </span>
      <input
        alt="text"
        class="music-toggle-button"
        src="icon-genre.svg"
        type="image"
      />
    </div>
    <div
      class="music-container"
    >
      <div
        class="song-card"
        data-testid="song-card"
      >
        <img
          alt="song"
          class="song-image"
          src="https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
        />
        <div
          class="song-container"
        >
          <div>
            <div
              class="song-name"
            >
              Aankh Marey (From Simmba)
            </div>
            <div
              class="song-artist"
            >
              Neha Kakkar
            </div>
          </div>
          <div
            class="song-image-container"
          >
            <input
              alt="text"
              class="song-heart-button"
              src="heart-gray.svg"
              type="image"
            />
            <div
              class="song-likes"
            >
              <span
                class="song-like-count"
              >
                0
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="song-card"
        data-testid="song-card"
      >
        <img
          alt="song"
          class="song-image"
          src="https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8"
        />
        <div
          class="song-container"
        >
          <div>
            <div
              class="song-name"
            >
              Gone, Gone, Gone
            </div>
            <div
              class="song-artist"
            >
              Phillip Phillips
            </div>
          </div>
          <div
            class="song-image-container"
          >
            <input
              alt="text"
              class="song-heart-button"
              src="heart-gray.svg"
              type="image"
            />
            <div
              class="song-likes"
            >
              <span
                class="song-like-count"
              >
                0
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
